consensus/src/creation/mod.rs

```diff
@@ -104,69 +126,89 @@ async fn create_unit<H: Hasher>(
 /// Section 5.1 for a discussion of this component.
 pub async fn run<H: Hasher>(
     conf: Config,
-    io: IO<H>,
+    mut io: IO<H>,
     mut starting_round: oneshot::Receiver<Option<Round>>,
     mut terminator: Terminator,
 ) {
+    futures::select! {
+        _ = read_starting_round_and_run_creator(conf, &mut io, &mut starting_round).fuse() =>
```

Comment:
isn't it better to pass just `starting_round` instead of a mut reference?

Reply:
This is due the `Terminator` - `starting_round` shouldn't be dropped until `terminator.terminate_sync().await` returns.